spec/Component.coffee

```diff
@@ -1573,3 +1573,79 @@ describe 'Component', ->
         ip = new noflo.IP 'data', 'foo'
         ip.scope = 'eh'
         sin1.post ip
+
+    describe 'using streams', ->
+      it 'should not trigger without a full stream without getting the whole stream', (done) ->
+        setTimeout ->
+          done()
+        , 1000
+        c = new noflo.Component
+          inPorts:
+            in:
+              datatype: 'string'
+          outPorts:
+            out:
+              datatype: 'string'
+          process: (input, output) ->
+            return unless input.hasStream 'in'
+            done new Error 'should never trigger this'
+        c.forwardBrackets = {}
+        c.inPorts.in.attach sin1
+        sin1.send 'eh'
```

Comment:
Please use `.post()` instead, the old socket methods will be deprecated at some point in near future.